integration-tests/src/tests/nearcore/sync_state_nodes.rs

```diff
@@ -395,3 +397,132 @@ fn sync_empty_state() {
         });
     });
 }
+
+/// Runs one node for some time, which dumps state to a temp directory.
+/// Start the second node which gets state parts from that temp directory.
+#[test]
+#[cfg_attr(not(feature = "expensive_tests"), ignore)]
+fn sync_state_dump() {
+    heavy_test(|| {
+        init_integration_logger();
+
+        let mut genesis = Genesis::test_sharded_new_version(
+            vec!["test1".parse().unwrap(), "test2".parse().unwrap()],
+            1,
+            vec![1, 1, 1, 1],
+        );
+        genesis.config.epoch_length = 10;
+
+        run_actix(async move {
+            let (port1, port2) =
+                (tcp::ListenerAddr::reserve_for_test(), tcp::ListenerAddr::reserve_for_test());
+            // Produce more blocks to make sure that state sync gets triggered when the second node starts.
+            let state_sync_horizon = 20;
+            let block_header_fetch_horizon = 1;
+            let block_fetch_horizon = 1;
+
+            let mut near1 = load_test_config("test1", port1, genesis.clone());
+            near1.client_config.min_num_peers = 0;
+            near1.client_config.min_block_production_delay = Duration::from_millis(200);
+            near1.client_config.max_block_production_delay = Duration::from_millis(400);
+            near1.client_config.epoch_sync_enabled = false;
+            let dump_dir = tempfile::Builder::new().prefix("state_dump_1").tempdir().unwrap();
+            near1.client_config.state_sync.dump = Some(DumpConfig {
+                location: Filesystem { root_dir: dump_dir.path().to_path_buf() },
+                restart_dump_for_shards: None,
+                iteration_delay: Some(Duration::from_millis(50)),
+            });
+
+            let dir1 = tempfile::Builder::new().prefix("sync_nodes_1").tempdir().unwrap();
+            let nearcore::NearNode {
+                view_client: view_client1,
+                state_sync_dump_handle: _state_sync_dump_handle,
+                ..
+            } = start_with_config(dir1.path(), near1).expect("start_with_config");
+            let dir2 = tempfile::Builder::new().prefix("sync_nodes_2").tempdir().unwrap();
+
+            let view_client2_holder = Arc::new(RwLock::new(None));
+            let arbiters_holder = Arc::new(RwLock::new(vec![]));
+            let arbiters_holder2 = arbiters_holder;
+
+            wait_or_timeout(100, 60000, || async {
+                if view_client2_holder.read().unwrap().is_none() {
+                    let view_client2_holder2 = view_client2_holder.clone();
+                    let arbiters_holder2 = arbiters_holder2.clone();
+                    let genesis2 = genesis.clone();
+                    let dir2 = dir2.path().clone();
```

Comment:
nit: not really needed, since you can just pass `dir2.path()` to `start_with_config()` below. same for `dump_dir`.